### ISSUE TITLE
Gemfire Samples throw IllegalStateException on Shutdown

### ISSUE THREAD

**Issue by rwinch** [spring-projects/spring-session#665]
If I run the following from the root project

NOTE: It does not happen if I run from the specific samples

```bash
$ gradlew clean check
```

I get the exception

```
:samples:httpsession-gemfire-clientserver:integrationTomcatStop
Invocation of destroy method failed on bean with name 'gemfirePool'
java.lang.IllegalStateException: Pool could not be destroyed because it is still in use by 1 regions
	at com.gemstone.gemfire.cache.client.internal.PoolImpl.destroy(PoolImpl.java:501)
	at org.springframework.data.gemfire.client.PoolFactoryBean.destroy(PoolFactoryBean.java:158)
	at org.springframework.beans.factory.support.DisposableBeanAdapter.destroy(DisposableBeanAdapter.java:272)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroyBean(DefaultSingletonBeanRegistry.java:578)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingleton(DefaultSingletonBeanRegistry.java:554)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingleton(DefaultListableBeanFactory.java:954)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingletons(DefaultSingletonBeanRegistry.java:523)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingletons(DefaultListableBeanFactory.java:961)
	at org.springframework.context.support.AbstractApplicationContext.destroyBeans(AbstractApplicationContext.java:1033)
	at org.springframework.context.support.AbstractApplicationContext.doClose(AbstractApplicationContext.java:1009)
	at org.springframework.context.support.AbstractApplicationContext.close(AbstractApplicationContext.java:961)
	at org.springframework.web.context.ContextLoader.closeWebApplicationContext(ContextLoader.java:581)
	at org.springframework.web.context.ContextLoaderListener.contextDestroyed(ContextLoaderListener.java:116)
	at org.apache.catalina.core.StandardContext.listenerStop(StandardContext.java:5061)
	at org.apache.catalina.core.StandardContext.stopInternal(StandardContext.java:5719)
	at org.apache.catalina.util.LifecycleBase.stop(LifecycleBase.java:232)
	at org.apache.catalina.Lifecycle$stop$0.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.apache.catalina.Lifecycle$stop$0.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callSafe(AbstractCallSite.java:80)
	at com.bmuschko.gradle.tomcat.embedded.BaseTomcatServerImpl.stop(BaseTomcatServerImpl.groovy:61)
	at com.bmuschko.gradle.tomcat.embedded.TomcatServer$stop$8.call(Unknown Source)
	at com.bmuschko.gradle.tomcat.internal.ShutdownMonitor.run(ShutdownMonitor.groovy:75)
```

The tests still pass, but this error should not be happening

**Comment by jxblum** [comment]
@rwinch -

I just got an error trying to run a build with the latest changes on `master`...

{code}
$ gradle clean build install

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/jblum/pivdev/spring-session/build.gradle' line: 16

* What went wrong:
A problem occurred evaluating root project 'spring-session-build'.
> No signature of method: build_69cls1hbl0kuqjlhk4d6u7vu73$_run_closure2.id() is applicable for argument types: (java.lang.String) values: [org.sonarqube]
  Possible solutions: is(java.lang.Object), is(java.lang.Object), find(), find(), find(groovy.lang.Closure), find(groovy.lang.Closure)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
{code}

**Comment by jxblum** [comment]
@rwinch - well, cannot reproduce it now... must have been a problem with `Gradle` trying to resolve its dependencies and plugins, argh!

**Comment by jxblum** [comment]
Fixed and committed to `master` in revision [dd3a571](https://github.com/spring-projects/spring-session/commit/dd3a57149477274cc6fca7e9e054eefb87671a28) for _Spring Session_ **1.3 M3**.

**Comment by rwinch** [comment]
Thanks John!
